File: solutions/804. Unique Morse Code Words.cpp
class Solution {
public:
    int uniqueMorseRepresentations(vector<string>& words) {
        vector<string> ans_arr;
        vector<string> mapping = {".-","-...","-.-.","-..",".","..-.","--.","....","..",".---","-.-",".-..","--","-.","---",".--.","--.-",".-.","...","-","..-","...-",".--","-..-","-.--","--.."};
        unordered_set<string> s;
        for (auto i:words) {
            string ans;
            for (auto j:i) ans+=mapping[j-'a'];
            s.insert(ans);
        }
        return s.size();
    }
};

File: leetcode/Algorithms/804. Unique Morse Code Words.py
class Solution:
    def uniqueMorseRepresentations(self, words: List[str]) -> int:
        ans_arr = []
        mapping = [".-","-...","-.-.","-..",".","..-.","--.","....","..",".---","-.-",".-..","--","-.","---",".--.","--.-",".-.","...","-","..-","...-",".--","-..-","-.--","--.."]
        for i in words:
            ans_words = ''
            for j in i:
                ans_words+= mapping[ord(j)-97]
            ans_arr.append(ans_words)
        return len(set(ans_arr))